crates/ibc/src/mock/context/applications.rs

```diff
@@ -11,7 +10,9 @@ use crate::applications::transfer::PrefixedCoin;
 use crate::core::ics24_host::identifier::{ChannelId, PortId};
 use crate::Signer;
 
-impl TokenTransferValidationContext for MockContext {
+pub struct MockTokenTransferModule;
+
+impl TokenTransferValidationContext for MockTokenTransferModule {
```

Comment:
Following our discussion on Module vs Application, it makes more sense if `TokenTrafer*` contexts be implemented on the same struct that implements `Module`. The same as in [basecoin](https://github.com/informalsystems/basecoin-rs/blob/main/src/modules/ibc/transfer.rs#L67).
We can impl them for `DummyTransferModule` under the `test_utils.rs`